docs/companion.md

```diff
@@ -3,3 +3,716 @@ sidebar_position: 4
+If `false`, files will be fully downloaded first, then uploaded. Defaults to `false`,
+but we recommended enabling it, especially if you’re expecting to upload large files.
+In future versions the default might change to `true`.
+
+#### `maxFileSize` `COMPANION_MAX_FILE_SIZE`
+
+If this value is set, companion will limit the maximum file size to process.
+If unset, it will process files without any size limit (this is the default).
+
+#### `periodicPingUrls` `COMPANION_PERIODIC_PING_URLS`
+
+If this value is set, companion will periodically send POST requests to the specified URLs.
+Useful for keeping track of companion instances as a keep-alive.
+
+#### `periodicPingInterval` `COMPANION_PERIODIC_PING_INTERVAL`
+
+Interval for periodic ping requests (in ms).
+
+#### `periodicPingStaticPayload` `COMPANION_PERIODIC_PING_STATIC_JSON_PAYLOAD`
+
+A `JSON.stringify`-able JavaScript Object that will be sent as part of the JSON body in the period ping requests.
+
+#### `allowLocalUrls` `COMPANION_ALLOW_LOCAL_URLS`
+
+A boolean flag to tell Companion whether to allow requesting local URLs (non-internet IPs).
+
+:::caution
+Only enable this in development. **Enabling it in production is a security risk.**
+:::
+
+#### `corsOrigins` `COMPANION_CLIENT_ORIGINS`
+
+Allowed CORS Origins (default `true`). 
+Passed as the `origin` option in [cors](https://github.com/expressjs/cors#configuration-options))
+
+#### `COMPANION_CLIENT_ORIGINS_REGEX`
+
+Like COMPANION_CLIENT_ORIGINS, but allows a single regex instead.
+`COMPANION_CLIENT_ORIGINS` will be ignored if this is used.
+
+#### `chunkSize` `COMPANION_CHUNK_SIZE`
+
+Controls how big the uploaded chunks are for AWS S3 Multipart and Tus.
+Smaller values lead to more overhead, but larger values lead to slower retries in case of bad network connections.
+Passed to tus-js-client [`chunkSize`](https://github.com/tus/tus-js-client/blob/master/docs/api.md#chunksize) as well as [AWS S3 Multipart](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/S3.html) `partSize`.
+
+### Standalone only options
+
+These environment variables apply only to Companion running in [standalone mode](#standalone).
+
+#### `COMPANION_COOKIE_DOMAIN`
+
+Allows you to customize the domain of the cookies created for Express sessions.
+
+#### `COMPANION_ONEDRIVE_DOMAIN_VALIDATION`
+
+Settings this variable to `true` enables a route that can be used to validate your app with OneDrive. Only set this if you are sure that you are also setting the correct value for `COMPANION_ONEDRIVE_KEY` (i.e application ID - **not** the secret!)
+
+#### `COMPANION_HIDE_WELCOME`
+
+Setting this to `true` disables the welcome message shown at `/`.
+
+### Options `_FILE` suffix
+
+Many environment variables can alternatively be specified using a `_FILE` suffix. In this case the secret will be loaded from a file instead of being specified directly on the command line. This can be useful to prevent accidentally leaking secrets.
```

Comment:
```suggestion
Many environment variables can alternatively be specified using a `_FILE` suffix. In this case the secret will be loaded from a file instead of being specified directly on the command line. This can be useful to prevent accidentally leaking secrets. The should be an absolute path to a file with any extension which only contains the secret, nothing else.
```